api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -130,22 +148,32 @@ public Workspace get(String ns, String firecloudName) {
         .collect(Collectors.toList());
   }
 
+  @Transactional
   @Override
-  public List<WorkspaceResponse> getWorkspaces() {
-    return getWorkspacesAndPublicWorkspaces().stream()
-        .filter(
-            workspaceResponse ->
-                !(workspaceResponse.getAccessLevel() != WorkspaceAccessLevel.OWNER
-                    && workspaceResponse.getAccessLevel() != WorkspaceAccessLevel.WRITER
-                    && workspaceResponse.getWorkspace().getPublished()))
-        .collect(Collectors.toList());
-  }
+  public WorkspaceResponse getWorkspace(String workspaceNamespace, String workspaceId) {
+    Workspace dbWorkspace = getRequired(workspaceNamespace, workspaceId);
 
-  @Override
-  public List<WorkspaceResponse> getPublishedWorkspaces() {
-    return getWorkspacesAndPublicWorkspaces().stream()
-        .filter(workspaceResponse -> workspaceResponse.getWorkspace().getPublished())
-        .collect(Collectors.toList());
+    org.pmiops.workbench.firecloud.model.WorkspaceResponse fcResponse;
+    org.pmiops.workbench.firecloud.model.Workspace fcWorkspace;
+
+    WorkspaceResponse response = new WorkspaceResponse();
```

Comment:
nit: I would almost want to see the fully qualified name for this `WorkspaceResponse` here as well to make it explicit that there are two of them now. IntelliJ might fight you on that though.